### ISSUE TITLE
Broken compatibility with MySQL 5.1 

### ISSUE THREAD

**Issue by TheMP** [metabase/metabase#3502]
The newly permissions_revision table contains remarks that exceed 60 characters. This breaks the migration from v0.19 to v0.20 with the error:

> Comment for table 'permissions_revision' is too long (max = 60)

This issue with MySQL was fixed after version 5.1 - however, still a lot of enterprise software is using 5.1.

**Comment by camsaul** [comment]
Fixed by #3509

**Comment by shakkhar** [comment]
Looks like this issue has resurfaced. When launching metabase.jar (latest stable, 0.25.2) against MySql 5.1.66 I get the following error and stack trace:

```
# MB_DB_DBNAME=metabase MB_DB_TYPE=mysql MB_DB_PORT=3306 MB_DB_USER=<username> MB_DB_PASS=<password> MB_DB_HOST=<host> java -jar metabase-0.25.2.jar
09-15 18:51:15 INFO metabase.util :: Loading Metabase...
09-15 18:51:22 INFO util.encryption :: DB details encryption is DISABLED for this Metabase instance. 🔓
09-15 18:51:31 INFO metabase.core :: Starting Metabase in STANDALONE mode
09-15 18:51:31 INFO metabase.core :: Launching Embedded Jetty Webserver with config:
 {:port 3000}

09-15 18:51:31 INFO metabase.core :: Starting Metabase version v0.25.2 (35bce2d release-0.25.2) ...
09-15 18:51:31 INFO metabase.core :: System timezone is 'Etc/Zulu' ...
09-15 18:51:31 DEBUG metabase.driver :: Registered driver :bigquery 🚚
09-15 18:51:31 DEBUG metabase.driver :: Registered driver :crate 🚚
09-15 18:51:31 DEBUG metabase.driver :: Registered driver :druid 🚚
09-15 18:51:31 WARN metabase.driver :: No -init-driver function found for 'metabase.driver.google'
09-15 18:51:31 DEBUG metabase.driver :: Registered driver :googleanalytics 🚚
09-15 18:51:32 DEBUG metabase.driver :: Registered driver :h2 🚚
09-15 18:51:32 DEBUG metabase.driver :: Registered driver :mongo 🚚
09-15 18:51:32 DEBUG metabase.driver :: Registered driver :mysql 🚚
09-15 18:51:32 DEBUG metabase.driver :: Registered driver :postgres 🚚
09-15 18:51:32 DEBUG metabase.driver :: Registered driver :presto 🚚
09-15 18:51:32 DEBUG metabase.driver :: Registered driver :redshift 🚚
09-15 18:51:32 DEBUG metabase.driver :: Registered driver :sqlite 🚚
09-15 18:51:32 DEBUG metabase.driver :: Registered driver :sqlserver 🚚
09-15 18:51:32 INFO metabase.core :: Setting up and migrating Metabase DB. Please sit tight, this may take a minute...
09-15 18:51:32 INFO metabase.db :: Verifying mysql Database Connection ...
09-15 18:51:32 INFO metabase.db :: Verify Database Connection ...  ✅
09-15 18:51:32 INFO metabase.db :: Running Database Migrations...
09-15 18:51:32 INFO metabase.db :: Setting up Liquibase...
09-15 18:51:32 INFO metabase.db :: Liquibase is ready.
09-15 18:51:32 INFO metabase.db :: Checking if Database has unrun migrations...
09-15 18:51:36 WARN liquibase :: modifyDataType will lose primary key/autoincrement/not null settings for mysql.  Use <sql> and re-specify all configuration if this is the case
09-15 18:51:36 WARN liquibase :: modifyDataType will lose primary key/autoincrement/not null settings for mysql.  Use <sql> and re-specify all configuration if this is the case
09-15 18:51:36 INFO metabase.db :: Database has unrun migrations. Waiting for migration lock to be cleared...
09-15 18:51:36 INFO metabase.db :: Migration lock is cleared. Running migrations...
09-15 18:51:36 WARN liquibase :: modifyDataType will lose primary key/autoincrement/not null settings for mysql.  Use <sql> and re-specify all configuration if this is the case
09-15 18:51:36 WARN liquibase :: modifyDataType will lose primary key/autoincrement/not null settings for mysql.  Use <sql> and re-specify all configuration if this is the case
java.sql.SQLException: Comment for table 'collection' is too long (max = 60)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:963)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3966)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3902)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2526)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2673)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2549)
	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1861)
	at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2073)
	at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2009)
	at com.mysql.jdbc.PreparedStatement.executeLargeUpdate(PreparedStatement.java:5098)
	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:1994)
	at clojure.java.jdbc$db_do_execute_prepared_statement$fn__10867.invoke(jdbc.clj:784)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:637)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:598)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at clojure.java.jdbc$db_do_execute_prepared_statement.invokeStatic(jdbc.clj:783)
	at clojure.java.jdbc$db_do_execute_prepared_statement.invoke(jdbc.clj:778)
	at clojure.java.jdbc$db_do_prepared.invokeStatic(jdbc.clj:814)
	at clojure.java.jdbc$db_do_prepared.invoke(jdbc.clj:795)
	at clojure.java.jdbc$execute_BANG_$execute_helper__10910.invoke(jdbc.clj:961)
	at clojure.java.jdbc$execute_BANG_.invokeStatic(jdbc.clj:963)
	at clojure.java.jdbc$execute_BANG_.invoke(jdbc.clj:943)
	at clojure.java.jdbc$execute_BANG_.invokeStatic(jdbc.clj:954)
	at clojure.java.jdbc$execute_BANG_.invoke(jdbc.clj:943)
	at metabase.db$migrate_up_if_needed_BANG_.invokeStatic(db.clj:164)
	at metabase.db$migrate_up_if_needed_BANG_.invoke(db.clj:151)
	at metabase.db$migrate_BANG_$fn__19465.invoke(db.clj:247)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:613)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:629)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:598)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at metabase.db$migrate_BANG_.invokeStatic(db.clj:235)
	at metabase.db$migrate_BANG_.invoke(db.clj:218)
	at metabase.db$run_schema_migrations_BANG_.invokeStatic(db.clj:365)
	at metabase.db$run_schema_migrations_BANG_.invoke(db.clj:360)
	at metabase.db$setup_db_BANG_.invokeStatic(db.clj:383)
	at metabase.db$setup_db_BANG_.doInvoke(db.clj:376)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at metabase.core$init_BANG_.invokeStatic(core.clj:103)
	at metabase.core$init_BANG_.invoke(core.clj:82)
	at metabase.core$start_normally.invokeStatic(core.clj:187)
	at metabase.core$start_normally.invoke(core.clj:181)
	at metabase.core$_main.invokeStatic(core.clj:279)
	at metabase.core$_main.doInvoke(core.clj:274)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at metabase.core.main(Unknown Source)
09-15 18:53:00 ERROR metabase.core :: Metabase Initialization FAILED:  Comment for table 'collection' is too long (max = 60)
09-15 18:53:00 INFO metabase.core :: Metabase Shutting Down ...
09-15 18:53:00 INFO metabase.core :: Metabase Shutdown COMPLETE
```

A custom build with all the long remarks stripped solves the problem.

**Comment by camsaul** [comment]
@shakkhar what version did you upgrade from?

**Comment by shakkhar** [comment]
Not sure I understand what you mean. This happened against an empty database. The last released Metabase version which works for me is 0.21.1.

**Comment by camsaul** [comment]
@shakkhar got it. I'll take a look, this should be easy enough to fix.
